src/python_testing/TC_BOOLCFG_4_4.py

```diff
@@ -0,0 +1,234 @@
+            TestStep("2a", "Read FeatureMap attribute"),
+            TestStep("2b", "Read AttributeList attribute"),
+            TestStep(3, "Verify AlarmsEnabled is supported"),
+            TestStep(4, "Create enabledAlarms and set to 0"),
+            TestStep("5a", "Enable VIS alarm in enabledAlarms"),
+            TestStep("5b", "Enable AUD alarm in enabledAlarms"),
+            TestStep("5c", "Set AlarmsEnabled attribute to value of enabledAlarms using AlarmsToEnableDisable command"),
+            TestStep(6, "Send TestEventTrigger with SensorTrigger event"),
+            TestStep(7, "Read AlarmsActive attribute"),
+            TestStep(8, "Verify VIS alarm is active"),
+            TestStep("9a", "Disable VIS alarm in enabledAlarms"),
+            TestStep("9b", "Set AlarmsEnabled attribute to value of enabledAlarms using AlarmsToEnableDisable command"),
+            TestStep(10, "Read AlarmsActive attribute"),
+            TestStep(11, "Verify AUD alarm is active"),
+            TestStep("12a", "Disable VIS alarm in enabledAlarms"),
+            TestStep("12b", "Set AlarmsEnabled attribute to value of enabledAlarms using AlarmsToEnableDisable command"),
+            TestStep(13, "Read AlarmsActive attribute"),
+            TestStep(14, "Send TestEventTrigger with SensorUntrigger event"),
+        ]
+        return steps
+
+    def pics_TC_BOOLCFG_4_4(self) -> list[str]:
+        pics = [
+            "BOOLCFG.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_BOOLCFG_4_4(self):
+
+        asserts.assert_true('PIXIT.BOOLCFG.TEST_EVENT_TRIGGER_KEY' in self.matter_test_config.global_test_params,
+                            "PIXIT.BOOLCFG.TEST_EVENT_TRIGGER_KEY must be included on the command line in "
+                            "the --int-arg flag as PIXIT.BOOLCFG.TEST_EVENT_TRIGGER_KEY:<key>")
+
+        endpoint = self.user_params.get("endpoint", 1)
+        enableKey = self.matter_test_config.global_test_params['PIXIT.BOOLCFG.TEST_EVENT_TRIGGER_KEY'].to_bytes(16, byteorder='big')
+
+        self.step(1)
+        attributes = Clusters.BooleanStateConfiguration.Attributes
+
+        self.step("2a")
+        feature_map = await self.read_boolcfg_attribute_expect_success(endpoint=endpoint, attribute=attributes.FeatureMap)
+
+        is_vis_feature_supported = feature_map & Clusters.BooleanStateConfiguration.Bitmaps.Feature.kVisual
+        is_aud_feature_supported = feature_map & Clusters.BooleanStateConfiguration.Bitmaps.Feature.kAudible
+
+        self.step("2b")
+        attribute_list = await self.read_boolcfg_attribute_expect_success(endpoint=endpoint, attribute=attributes.AttributeList)
+
+        self.step(3)
+        if attributes.AlarmsEnabled.attribute_id not in attribute_list:
+            logging.info("AlarmsEnabled not supported skipping test case")
+
+            # Skipping all remainig steps
+            self.step(4)
```

Comment:
as above, you can do this in a for loop and then it auto-expands if you need to update the test.